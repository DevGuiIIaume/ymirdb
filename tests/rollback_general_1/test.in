set a 1
set b a 2
set c 3
push a c
set d 4
set d 4 a b c
forward a
forward b
forward c
forward d
backward a
backward b
backward c
backward d
snapshot
set e d 5
snapshot
checkout 1
forward a
forward b
forward c
forward d
forward e
backward a
backward b
backward c
backward d
backward e
checkout 2
forward a
forward b
forward c
forward d
backward a
backward b
backward c
backward d
forward e
backward e
del e
del d
del b
del a
del c
list entries
rollback 1
list snapshots
forward a
forward b
forward c
forward d
backward a
backward b
backward c
backward d
bye
