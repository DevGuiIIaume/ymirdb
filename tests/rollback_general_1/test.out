> ok

> ok

> ok

> ok

> ok

> ok

> c

> a, c

> nil

> a, b, c

> b, d

> d

> a, b, d

> nil

> saved as snapshot 1

> ok

> saved as snapshot 2

> ok

> c

> a, c

> nil

> a, b, c

> no such key

> b, d

> d

> a, b, d

> nil

> no such key

> ok

> c

> a, c

> nil

> a, b, c

> b, d, e

> d, e

> a, b, d, e

> e

> a, b, c, d

> nil

> ok

> ok

> ok

> ok

> ok

> no entries

> ok

> 1

> c

> a, c

> nil

> a, b, c

> b, d

> d

> a, b, d

> nil

> bye
